File: ListView1/app/src/main/java/com/example/listview1/MainActivity.java
package com.example.listview1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
ListView listView;
    String[] arr={"tcs","infosys","wipro","hcl","tech mahindra","mindtree","l&t infotech","happiest minds"};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView=findViewById(R.id.list1);
        ArrayAdapter ad=new ArrayAdapter(this, android.R.layout.simple_expandable_list_item_1,arr);
     listView.setAdapter(ad);
     listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
         @Override
         public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
             Toast.makeText(MainActivity.this,"you clicked on "+id,Toast.LENGTH_LONG).show();
         }
     });
    }
}

File: ListAdapter/app/src/main/java/com/example/listadapter/MainActivity.kt
package com.example.listadapter

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import java.util.*
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val listView : ListView =findViewById(R.id.listView)
        var array = ArrayList<String>(Arrays.asList("tcs","hcl","wipro","infosys"))
       val customListAdapter:CustomListAdapter  = CustomListAdapter(this,array)
        listView.adapter = customListAdapter
    }
}